pandas/io/parquet.py

```diff
@@ -92,7 +97,7 @@ def write(
         **kwargs,
     ):
         self.validate_dataframe(df)
-        path, _, _, should_close = get_filepath_or_buffer(path, mode="wb")
+        file_obj, _, _, should_close = get_filepath_or_buffer(path, mode="wb")
```

Comment:
you didn't change `path` to `file_obj` in the ` if partition_cols is not None:`  block. Was that on purpose?

Reply:
That was indeed on purpose, `write_to_dataset` doesn't support a file like object when that path is a directory.


```
import pyarrow.parquet 
import pandas as pd
from pandas.io.common import get_filepath_or_buffer

path = "s3://pandas-test/dev"
file_obj, _,_,_, = get_filepath_or_buffer(path)
df = pd.DataFrame({"a": [1,2], "b": [3,4]})
table = pyarrow.Table.from_pandas(df) 

# Works
pyarrow.parquet.write_to_dataset(table, path, partition_cols=["a"])

# Throws AttributeError: 'NoneType' object has no attribute '_isfilestore
pyarrow.parquet.write_to_dataset(table, file_obj, partition_cols=["a"])
```  
@jorisvandenbossche

Reply:
OK, can you add some clarifying comments for it?

Reply:
Sure done!

Reply:
is this *always* a file_obj, *never* a path? e.g. should rename to filepath_or_buffer ?

Reply:
Good point, i've renamed to ```file_obj_or_path``` since when a local path is passed in a path str  is returned.